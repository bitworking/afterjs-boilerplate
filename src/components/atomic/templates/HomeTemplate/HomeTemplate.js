// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import HomeHeader from '../../molecules/HomeHeader/HomeHeader';

type Props = {
  /** please describe the prop children */
  children?: Node,
}

const HomeTemplateStyled = styled.div`
  text-align: center;
`;

const HomeIntro = styled.p`
  font-size: large;
`;

const HomeTemplate = (props: Props): Node => (
  <HomeTemplateStyled>
    <HomeHeader />
    <HomeIntro>
      To get started, edit <code>src/Home.js</code> or{' '}
      <code>src/About.js</code>and save to reload.
    </HomeIntro>
    <Link to="/about">About</Link>
  </HomeTemplateStyled>
);

HomeTemplate.defaultProps = {

}; 

export default HomeTemplate;
