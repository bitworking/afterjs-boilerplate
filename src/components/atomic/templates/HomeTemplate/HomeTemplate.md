## Demo of the template HomeTemplate

    <HomeTemplate

    >
       Lorem Ipsum Children
    </HomeTemplate>
    
Please add more examples    
