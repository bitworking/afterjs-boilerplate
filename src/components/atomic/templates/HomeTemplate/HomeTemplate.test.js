import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {MemoryRouter} from 'react-router-dom';

import HomeTemplate from './HomeTemplate';

it('renders default HomeTemplate snapshot', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <HomeTemplate
  
      >
         Lorem Ipsum Children
      </HomeTemplate>
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
