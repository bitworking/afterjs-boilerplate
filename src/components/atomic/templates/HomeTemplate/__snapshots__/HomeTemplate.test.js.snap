// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default HomeTemplate snapshot 1`] = `
.c1 {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.c1 img {
  -webkit-animation: logo-spin infinite 20s linear;
  animation: logo-spin infinite 20s linear;
  height: 80px;
}

.c0 {
  text-align: center;
}

.c2 {
  font-size: large;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="logo"
      src="react.svg"
    />
    <h2>
      Welcome to After.js
    </h2>
  </div>
  <p
    className="c2"
  >
    To get started, edit 
    <code>
      src/Home.js
    </code>
     or
     
    <code>
      src/About.js
    </code>
    and save to reload.
  </p>
  <a
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
</div>
`;
